Microsoft.DotNet.Interactive/KernelBase.cs

```diff
@@ -251,7 +256,21 @@ public void RegisterForDisposal(IDisposable disposable)
             IKernelCommand command,
             KernelInvocationContext context) => context.HandlingKernel = this;
 
-        public void Dispose() => _disposables.Dispose();
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposables.IsDisposed)
+            {
+                return;
+            }
+
+            _disposables.Dispose();
```

Comment:
This should only be called when `disposing` is `true`. When you are finalizing `KernelBase`, you shouldn't be touching other managed objects. 

See https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose?view=netframework-4.8#the-disposeboolean-overload

Reply:
Yes that is right, but we are disposing unmanaged bits in this since we register disposals on inherited classes such as FSharpKernel which will register to dispose the generated assemblies.

Reply:
FSharpKernel arguably should be following the correct dispose pattern, which I know for a fact we don't :)